proto/p4/p4runtime.proto

```diff
@@ -136,6 +136,16 @@ message TableEntry {
   // this is stored, as long as it is returned with the rest of the entry in
   // a Read RPC.
   uint64 controller_metadata = 5;
+  // meter_config and counter_data are convenience fields that enable the
+  // controller to configure the direct resources associated with the table at
+  // the same time as a match-action entry is inserted or modified. When
+  // updating a table entry, leaving these fields unset means that the direct
+  // resources (if any) will not be updated. When reading a table entry, the
+  // values associated with the direct resources (if any) will be returned. If
+  // the table does not have a direct resource, then setting the direct resource
```

Comment:
Instead of "If the table does not have a direct resource" I would say "If the table does not support direct resources"